aptos-move/aptos-aggregator/src/delayed_field_extension.rs

```diff
@@ -40,6 +41,12 @@ impl DelayedFieldData {
         input: SignedU128,
         resolver: &dyn DelayedFieldResolver,
     ) -> PartialVMResult<bool> {
+        if resolver.is_delayed_field_optimization_capable() {
+            fail_point!(
```

Comment:
Seems that this way we end up with a branch in real code? I think you can add a boolean argument to the fail macro to make it execute conditionally:

https://docs.rs/fail/latest/fail/macro.fail_point.html

Not sure though if it is evaluated lazily or not